src/python/pants/python/python_setup.py

```diff
@@ -274,17 +279,38 @@ def get_pex_python_paths():
             return []
 
     @staticmethod
-    def get_pyenv_paths(pyenv_root_func=None):
+    def get_pyenv_paths(pyenv_root_func=None, pyenv_local=False):
```

Comment:
A good practice for boolean fields is to enforce that this must be used with keyword arguments, meaning this will fail `cls.get_pyenv_paths(pyenv_root_func, True)`.

```suggestion
    def get_pyenv_paths(*, pyenv_root_func=None, pyenv_local=False):
```

The star means "everything after this star, you must use a keyword arg (kwarg) rather than a positiona arg".

Reply:
Another super helpful thing would be to add type hints to this function. We've been adding them incrementally to the code base to improve the quality of the code. 

```suggestion
    def get_pyenv_paths(*, pyenv_root_func: Optional[Callable] = None, pyenv_local: bool = False) -> List[str]:
```

You'll need to add `from typing import Callable, List, Optional` for this to work. Make sure to run `build-support/githooks/pre-commit` (or `build-support/bin/mypy.py`) to ensure MyPy is happy.

Reply:
Nice to know it, thanks 👍